iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergMergeRecordReader.java

```diff
@@ -87,6 +97,12 @@ private CloseableIterable<T> openGeneric(ContentFile contentFile, Schema readSch
               getTable().io().newInputFile(dataFile.path().toString()),
               dataFile.keyMetadata()));
       schema = readSchema;
+    } else if (contentFile instanceof DeleteFile) {
+      DeleteFile deleteFile = (DeleteFile) contentFile;
```

Comment:
can't we cast to `ContentFile` file and reuse bigger part, just amend the schema?

Reply:
Done.